prometheus_client/exposition.py

```diff
@@ -102,14 +103,30 @@ def do_GET(self):
     def log_message(self, format, *args):
         """Log nothing."""
 
+    @staticmethod
+    def factory(registry):
+        """Returns a dynamic MetricsHandler class tied
+           to the passed registry.
+        """
+        # This implementation relies on MetricsHandler.registry
+        #  (defined above and defaulted to core.REGISTRY).
+
+        # As we have unicode_literals, we need to create a str()
+        #  object for type().
+        cls_name = str('MetricsHandler')
+        MyMetricsHandler = type(cls_name, (MetricsHandler, object),
```

Comment:
The base class should be BaseHTTPRequestHandler

Reply:
When `type()` is invoked with three arguments, it returns a new-style class.

`BaseHTTPRequestHandler`:
  -  is still in the mro 
  -  is an old-style class and it requires the `object` to be added to the hierarchy.

```
# Create two old-style classes.
class OldStyle: pass
class Child(OldStyle): pass
# a new-style class can't have only classic bases
MyChild = type('MyChild', (Child,), {"a":1}) #  TypeError: 

# Add object to bases.
MyChild = type('MyChild', (Child,object), {"a":1})
# No need to add all the chain.
issubclass(MyChild, OldStyle)
MyChild.mro()

```